src/os/include/IConditionVariable.h

```diff
@@ -0,0 +1,23 @@
+#ifndef __ICONDITIONVARIABLE_H_
+#define __ICONDITIONVARIABLE_H_
+
+#include <cstdint>
+
+/**@brief Conditin variable, can notify and wait on data
+ *Only one thread can wait at the time */
+class IConditionVariable {
+  public:
+    virtual ~IConditionVariable() = default;
+
+    /**@brief notify the waiting thread, see the wait function*/
+    virtual void notify() = 0;
```

Comment:
If we ever plan to use this class in the bsp layers with freertos, we will need a notifyFromIsr method. We can add it later if we want to migrate the code. You could just add a TODO in the implementation for freertos for now.